sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala

```diff
@@ -67,8 +74,7 @@ case class AQEShuffleReadExec private(
             case e: Expression => r.updateAttr(e).asInstanceOf[Partitioning]
             case other => other
           }
-        case _ =>
-          throw new IllegalStateException("operating on canonicalization plan")
+        case _ => super.outputPartitioning
```

Comment:
it seems overkill to throw error and other canonicalized spark plan can also work